sqlobject/src/main/java/org/jdbi/v3/sqlobject/SqlMethodAnnotatedHandlerDecorator.java

```diff
@@ -0,0 +1,86 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.sqlobject;
+
+import static java.util.stream.Collectors.toList;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.ToIntFunction;
+import java.util.stream.Stream;
+
+/**
+ * Applies decorations to method handlers, according to any {@link SqlMethodDecoratingAnnotation decorating annotations}
+ * present on the method. If multiple decorating annotations are present, the order of application can be controlled
+ * using the {@link DecoratorOrder} annotation.
+ * <p>
+ * This decorator is registered out of the box.
+ * </p>
+ */
+public class SqlMethodAnnotatedHandlerDecorator implements HandlerDecorator {
+    @Override
+    public Handler decorateHandler(Handler base, Class<?> sqlObjectType, Method method) {
+        Handler handler = base;
+
+        List<Class<? extends Annotation>> annotationTypes = Stream.of(method.getAnnotations())
+                .map(Annotation::annotationType)
+                .filter(type -> type.isAnnotationPresent(SqlMethodDecoratingAnnotation.class))
+                .collect(toList());
+
+        Stream.of(method, sqlObjectType)
+                .map(e -> e.getAnnotation(DecoratorOrder.class))
+                .filter(Objects::nonNull)
+                .findFirst()
+                .ifPresent(order -> {
+                    annotationTypes.sort(createDecoratorComparator(order).reversed());
+                });
+
+        List<HandlerDecorator> decorators = annotationTypes.stream()
+                .map(type -> type.getAnnotation(SqlMethodDecoratingAnnotation.class))
+                .map(a -> buildDecorator(a.value()))
+                .collect(toList());
+
+        for (HandlerDecorator decorator : decorators) {
+            handler = decorator.decorateHandler(handler, sqlObjectType, method);
+        }
+
+        return handler;
+    }
+
+    private Comparator<Class<? extends Annotation>> createDecoratorComparator(DecoratorOrder order) {
+        List<Class<? extends Annotation>> ordering = Arrays.asList(order.value());
+
+        ToIntFunction<Class<? extends Annotation>> indexOf = type -> {
+            int index = ordering.indexOf(type);
+            return index == -1 ? ordering.size() : index;
+        };
+
+        return Comparator.comparingInt(indexOf::applyAsInt);
+    }
+
+    private static HandlerDecorator buildDecorator(Class<? extends HandlerDecorator> decoratorClass) {
+        HandlerDecorator decorator;
+        try {
+            decorator = decoratorClass.getConstructor().newInstance();
```

Comment:
We could the variable declaration and just use `return decoratorClass.getConstructor().newInstance();`